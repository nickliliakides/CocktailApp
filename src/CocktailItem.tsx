import React, { FC } from 'react';
import ICocktail from './Interfaces/ICocktail';

interface IProps {
  cocktail: ICocktail
}

const CocktailItem: FC<IProps> = ({ cocktail }): JSX.Element => {
  return (
    <li className='cocktail-item'>
      <div className='cocktail-image'>
        <img
          src={cocktail.img ? cocktail.img : 'img/no-image.png'}
          width={128}
          alt={cocktail.name ? cocktail.name : 'Unknown cocktail'}
        />
      </div>
      <div className='cocktail-details w-100 p-3 d-flex flex-column'>
        <h1 className='text-center'>{cocktail.name}</h1>
        <div className='d-flex'>
          <div className='instructions'>
            <p>Instructions:</p>
            <p>{cocktail.instructions}</p>
          </div>

          <div className='ingredients'>
            <p>Ingredients:</p>
            <ul>
              {cocktail.ingredients.map(
                (ing: string, i: number): JSX.Element => (
                  <li className='ingredient-item' key={i}>
                    <svg className='ingredient-icon'>
                      <use href='img/icons.svg#icon-check'></use>
                    </svg>
                    <p>{ing}</p>
                  </li>
                )
              )}
            </ul>
          </div>
        </div>
      </div>
    </li>
  );
};

export default CocktailItem;
