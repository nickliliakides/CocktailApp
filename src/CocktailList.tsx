import { FC } from 'react';
import ICocktail from './Interfaces/ICocktail';
import './CocktailList.styles.scss';
import CocktailItem from './CocktailItem';
import Loader from './Loader';

interface IProps {
  cocktails: ICocktail[]
  loading: boolean
}

const CocktailList: FC<IProps> = ({ cocktails, loading }): JSX.Element => {
  return loading ? (
    <Loader />
  ) : (
    <div className='list-container'>
      <ul>
        {cocktails &&
          cocktails?.map((cocktail: ICocktail) => (
            <CocktailItem key={cocktail.id} cocktail={cocktail} />
          ))}
      </ul>
    </div>
  );
};

export default CocktailList;
