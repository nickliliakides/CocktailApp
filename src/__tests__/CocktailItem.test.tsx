import { shallow, ShallowWrapper } from 'enzyme';
import CocktailItem from '../CocktailItem';
import { mockCocktails } from './CocktailList.test';

let wrapper: ShallowWrapper;

describe('<CocktailItem  />', () => {
  beforeEach(() => {
    wrapper = shallow(<CocktailItem cocktail={mockCocktails[0]} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should matches its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
