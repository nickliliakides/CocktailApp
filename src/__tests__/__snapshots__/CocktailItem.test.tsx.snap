// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CocktailItem  /> should matches its snapshot 1`] = `
<li
  className="cocktail-item"
>
  <div
    className="cocktail-image"
  >
    <img
      alt="test Name 1"
      src="/testurl1"
      width={128}
    />
  </div>
  <div
    className="cocktail-details w-100 p-3 d-flex flex-column"
  >
    <h1
      className="text-center"
    >
      test Name 1
    </h1>
    <div
      className="d-flex"
    >
      <div
        className="instructions"
      >
        <p>
          Instructions:
        </p>
        <p>
          test instructions 1
        </p>
      </div>
      <div
        className="ingredients"
      >
        <p>
          Ingredients:
        </p>
        <ul>
          <li
            className="ingredient-item"
            key="0"
          >
            <svg
              className="ingredient-icon"
            >
              <use
                href="img/icons.svg#icon-check"
              />
            </svg>
            <p>
              teat ingredient 1
            </p>
          </li>
          <li
            className="ingredient-item"
            key="1"
          >
            <svg
              className="ingredient-icon"
            >
              <use
                href="img/icons.svg#icon-check"
              />
            </svg>
            <p>
               1/2 test ingr
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</li>
`;

exports[`<CocktailList /> should matches its snapshot 1`] = `
<div
  className="list-container"
>
  <ul>
    <CocktailItem
      cocktail={
        Object {
          "id": "1",
          "img": "/testurl1",
          "ingredients": Array [
            "teat ingredient 1",
            " 1/2 test ingr",
          ],
          "instructions": "test instructions 1",
          "name": "test Name 1",
        }
      }
      key="1"
    />
    <CocktailItem
      cocktail={
        Object {
          "id": "2",
          "img": "/testurl2",
          "ingredients": Array [
            "teat ingredient 2",
            " 1/2 test ingr",
          ],
          "instructions": "test instructions 2",
          "name": "test Name 2",
        }
      }
      key="2"
    />
    <CocktailItem
      cocktail={
        Object {
          "id": "3",
          "img": "/testurl3",
          "ingredients": Array [
            "teat ingredient 3",
            " 1/2 test ingr",
          ],
          "instructions": "test instructions 3",
          "name": "test Name 3",
        }
      }
      key="3"
    />
  </ul>
</div>
`;
